### ISSUE TITLE
appcache links prefixed with /__browser.legacy/ in app.manifest

### ISSUE THREAD

**Issue by CaptainN** [meteor/meteor#9757]
This applies to ongoing work for modern/legacy bundles in 1.6.2

`appcache` is outputing all links to the manifest in the `CACHE:` and `FALLBACK:` sections prefixed with `/__browser.legacy/...`. This is incorrect, as many of those files will be linked directly from root - which means the cached assets will not be used by the user (most of the items which show in the cache manifest come from the `/public/` directory).

Additionally, the bottom section `NETWORK:` is not prefixed. This creates some double links. For example, I have `sw.js` in my `/public/` directory, and it shows up as both prefixed (in CACHE and FALLBACK) and unprefixed (in NETWORK).

The main js and css bundles should continue to output the legacy prefix, or based on the detected modern support, if that can be determined server side (I assume so, is that how modern bundles are sent?). If not, it should include the legacy bundle. Secondarily, whatever mechanism determines legacy/modern bundle delivery should match for setting the bundle in the app.manifest.

**Comment by CaptainN** [comment]
Looking into this, it looks like the manifest is iterating over the assets in `WebApp.clientPrograms[WebApp.defaultArch].manifest.forEach` to pull out the manifest links. It's easy enough to adjust this to output the default packages by changing `WebApp.defaultArch` to `'web.browser'`, however, I'm not sure that's enough to fix this on actual legacy browsers.

What mechanism does Meteor use to determine which bundle to send? That same logic should probably be used here. I'm happy to complete a PR if it's easy enough to accomplish.

**Comment by CaptainN** [comment]
@benjamn Actually, I found it. This works to fix up the modern browser use case:

```JavaScript
  var request = WebApp.categorizeRequest(req);
  var reqArch = isModern(request.browser) ? 'web.browser' : 'web.browser.legacy';
  WebApp.clientPrograms[reqArch].manifest.forEach(resource => {
```

I'll see about fixing up the legacy use case as well.

**Comment by CaptainN** [comment]
I took a stab at a PR #9776

**Comment by benjamn** [comment]
This should be fixed if you run `meteor update 1.7-beta.26`?

**Comment by CaptainN** [comment]
It works! Thanks - it looks like there was still quite a bit left to do even after my PR. I appreciate you finishing that up.
